src/Utopia/Messaging/Adapters/SMS/TextMagic.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Utopia\Messaging\Adapters\SMS;
+
+// Reference Material
+// https://www.textmagic.com/docs/api/send-sms/#How-to-send-bulk-text-messages
+
+use Utopia\Messaging\Messages\SMS;
+use Utopia\Messaging\Adapters\SMS\SMS as SMSAdapter;
+
+class TextMagic extends SMSAdapter
+{
+    /**
+     * @param string $username TextMagic account username
+     * @param string $apiKey TextMagic account API key
+     */
+    public function __construct(
+        private string $username,
+        private string $apiKey
+    ) {
+    }
+
+    public function getName(): string
+    {
+        return 'TextMagic';
+    }
+
+    public function getMaxMessagesPerRequest(): int
+    {
+        return PHP_INT_MAX;
```

Comment:
We will probably hit HTTP request size limit before that.

Reply:
In that case the exception will propagate to the caller, the service has no limit, unless we want to apply an arbitrary one ourselves instead?

Reply:
Not sure what might be the best approach, but we would need to take payload size to consideration at some point, either here or in Appwrite itself.